wee_slack.py

```diff
@@ -2639,6 +2725,63 @@ def process_message(message_json, eventrouter, store=True, download=True, **kwar
     if download:
         download_files(message_json, **kwargs)
 
+def process_subteam_created(subteam_json, eventrouter, **kwargs):
+    request_metadata = subteam_json['wee_slack_metadata']
+    team = eventrouter.teams[request_metadata.get('team')]
+    subteam_json_info = subteam_json['subteam']
+    subteam = SlackSubteam(team.identifier, **subteam_json_info)
+    team.subteams[subteam_json_info['id']] = subteam
+
+def process_subteam_updated(subteam_json, eventrouter, **kwargs):
+    request_metadata = subteam_json['wee_slack_metadata']
+    team = eventrouter.teams[request_metadata.get('team')]
+    new_subteam_info = subteam_json['subteam']
+    messages = []
+
+    current_subteam_info = team.subteams[new_subteam_info.get('id')]
+
+    if current_subteam_info.name != new_subteam_info['name']:
+        template = "{current_name} has updated its name to {new_name} in team {team}"
+        message = template.format(current_name=current_subteam_info.name,
+                new_name=new_subteam_info['name'], team=team.preferred_name)
+        messages.append(message)
+    if current_subteam_info.description != new_subteam_info['description']:
+        template = "{name} has updated its description to \"{description}\" in team {team}"
+        message = template.format(name=current_subteam_info.name, description=new_subteam_info['description'],
+                team=team.preferred_name)
+        messages.append(message)
+    if current_subteam_info.handle != new_subteam_info['handle']:
+        team.subteam_handles_to_id[new_subteam_info['handle']] = new_subteam_info.get('id')
+        template = '{name} has updated its handle to @{handle} in team {team}'
+        message = template.format(name=current_subteam_info.name, handle=new_subteam_info['handle'],
+                team=team.preferred_name)
+        messages.append(message)
+
+    team.subteams[new_subteam_info.get('id')] = SlackSubteam(team.identifier, **new_subteam_info)
+
+    for message in messages:
+        team.buffer_prnt(message, message=True)
```

Comment:
Fair enough. I wasn't certain myself so I added as much as I could just in case. I remove of the usergroups notification
events except when the usergroup handle changes. A user can enable or disable that option in the configuration. I think
it useful if the handle changes so you can keep using without interruptions.